### PR TITLE
Work

### PR DESCRIPTION
**Pull request by Eli-BB8**
roei create

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -6,8 +6,8 @@ COPY ["package.json", "package-lock.json*", "./"]
 RUN npm install --production=false --silent
```
Comment (Eli-BB8) [inline Dockerfile:6]: do somthing better

File: src/builds/buildDI.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import matchedRecordType from '../types/matchedRecord';
+import fieldNames from '../config/fieldNames';
+import cleanObj from '../utils/removeBlankAtt';
+import digitalIdentityObj from '../types/digitalIdentity';
+import assembleUniqueID from '../utils/assembleUniqueID';
+
+export default (record: matchedRecordType, identifier: string) => {
+    const digitalIdentity: digitalIdentityObj = {
+        type: record.source === fieldNames.sources.mir ? 'digUser' : 'domUser',
```
Comment (Eli-BB8) [inline src/builds/buildDI.ts:9]: fildname or else

File: src/initializeRabbit.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    await menash.connect(rabbit.uri);
+    await menash.declareQueue(rabbit.consumeQueue);
+    await menash.declareQueue(rabbit.produceQueue);
+    await menash.declareQueue(rabbit.logQueue);
+
+    console.log('Rabbit connected');
+
+    await menash.queue(rabbit.consumeQueue).activateConsumer(
+        async (msg: ConsumerMessage) => {
+            const record: matchedRecord = msg.getContent() as matchedRecord;
```
Comment (Eli-BB8) [inline src/initializeRabbit.ts:22]: tryCatch

File: src/types/mergedObject.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import matchedRecordType from './matchedRecord'
```
Comment (Eli-BB8) [inline src/types/mergedObject.ts:1]: fix or delete

File: src/utils/assembleRoleID.ts [hunk 5]
```diff
@@ -0,0 +1,8 @@
+import matchedRecordType from '../types/matchedRecord';
+import { ROLE_ID_SUFFIXES } from '../config/db_enums';
+
+const roleIdSuffixes: Map<string, string> = new Map<string, string>(JSON.parse(JSON.stringify(ROLE_ID_SUFFIXES)));
```
Comment (Eli-BB8) [inline src/utils/assembleRoleID.ts:4]: wtf

**Review by Eli-BB8 — COMMENTED** [review]
look emazing without a

File: src/buildROGD.ts [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+const { rabbit } = config;
+
+export default (record: matchedRecordType): void => {
+    const identifier: string = (record.identityCard || record.personalNumber || record.goalUserId)!;
+    const di: digitalIdentityObj = buildDI(record, identifier);
+    const role: roleObj | null = record.hierarchy && di.isRoleAttachable ? buildRole(record, di.uniqueId) : null;
+    const og: organizationGroupObj | null = role ? buildOG(record) : null;
+
+    menash.send(rabbit.produceQueue, { di, role, og });
```
Comment (NoamYes) [inline src/buildROGD.ts:19]: don't import menash, use service function

File: src/config/db_enums.ts [hunk 7]
```diff
@@ -0,0 +1,18 @@
+import fieldNames from './fieldNames';
+
+export const DOMAIN_SUFFIXES = [
```
Comment (NoamYes) [inline src/config/db_enums.ts:3]: remove, "move" to basic match

File: src/initializeRabbit.ts [hunk 8]
```diff
@@ -0,0 +1,31 @@
+/* eslint-disable no-console */
+/* eslint-disable import/prefer-default-export */
```
Comment (NoamYes) [inline src/initializeRabbit.ts:2]: remember to enable

File: src/types/mergedObject.ts (hunk 4 above)
Comment (NoamYes) [reply src/types/mergedObject.ts:1]: fix or delete

File: src/utils/assembleRoleID.ts (hunk 5 above)
Comment (NoamYes) [reply src/utils/assembleRoleID.ts]: wtf

**Review by NoamYes — COMMENTED** [review]
amazing, but you can do better.

**Review by 770menahem — APPROVED** [review]

**Pull request merged**